Pantheon use mutter334

worldofpeace opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Just opened for my testing and tracking.

Pantheon with latest mutter is almost stable, with only one real issue reported left
* https://github.com/elementary/gala/issues/636

and maybe two crashes that I need to debug.

There still needs to be a change for g-s-d 3.32 https://github.com/elementary/gala/issues/518 https://github.com/elementary/greeter/issues/382, but I believe almost everything should be working.

Lastly tracking a weird graphical glitch https://github.com/elementary/os-patches/issues/80, believe that's an issue in mutter which I've kept a patch for in 3.28 
*  https://github.com/NixOS/nixpkgs/blob/139806d89d77a3c59785e4e59c00c7186e8e9ea6/pkgs/desktops/gnome-3/core/mutter/0001-Revert-ClutterActor-Preserve-valid-paint-volumes-til.patch

If we can't get the fixed timely I might move the patch into `gnome3.mutter` proper.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

worldofpeace committed:
pantheon.elementary-gsettings-schemas: add gala override

jonringer reviewed (approved):
`nix-review` passes on NixOS
diff LGTM
commits LGTM
can't test, don't run compatible DE

```
[30 built, 120 copied (350.3 MiB), 61.3 MiB DL]
https://github.com/NixOS/nixpkgs/pull/73906
18 package were built:
pantheon.elementary-greeter pantheon.elementary-gsettings-schemas pantheon.elementary-session-settings pantheon.gala pantheon.switchboard-plug-pantheon-shell pantheon.switchboard-with-plugs pantheon.wingpanel pantheon.wingpanel-applications-menu pantheon.wingpanel-indicator-bluetooth pantheon.wingpanel-indicator-datetime pantheon.wingpanel-indicator-keyboard pantheon.wingpanel-indicator-network pantheon.wingpanel-indicator-nightlight pantheon.wingpanel-indicator-notifications pantheon.wingpanel-indicator-power pantheon.wingpanel-indicator-session pantheon.wingpanel-indicator-sound pantheon.wingpanel-with-indicators
```

worldofpeace committed:
pantheon.elementary-settings-daemon: 3.30.2 -> 3.34.1

jonringer commented:
@worldofpeace status?

worldofpeace commented:
These issues need to be fixed before I commit it to NixOS
* https://github.com/elementary/gala/issues/518
* https://github.com/elementary/greeter/issues/382
* https://github.com/elementary/gala/issues/636

then it's "sorta" stable and can be improved upon hopefully before our release.

worldofpeace commented:
Yay, everything seems to be fixed upstream and we just need to port over https://github.com/elementary/gala/pull/754 for https://github.com/elementary/greeter/issues/382

worldofpeace commented:
Hmm, they've ported everything to 3.36 too. This is looking good.

worldofpeace committed:
pantheon.elementary-notifications: init at 2020-04-04

worldofpeace committed:
pantheon.elementary-session-settings: 5.0.3 -> 2019-11-12

So we can drop the meson.patch intree.

worldofpeace committed:
pantheon.elementary-session-settings: add g-s-d version mappings

Needed to have a correct pantheon.session gnome-session file.

worldofpeace committed:
pantheon.gala: use-new-notifications default true

worldofpeace committed:
nixos/pantheon: use new notifications

We have this bug https://github.com/elementary/gala/issues/636
when using notifications in gala. It's likely to not really be fixed
because all development is on the new notifications server.

worldofpeace committed:
pantheon.elementary-default-settings: fix media key syntax

worldofpeace commented:
The greeter has had the new gala code ported over for media keys:
* https://github.com/elementary/greeter/commit/39cba2b462897dbbc10a994e9a8c7c39c4a95178
* https://github.com/elementary/greeter/commit/c24ff5e4aa422592f54928c083e0b060f7607184

And I discovered one last real bug, and opened PRs with fixes:
* https://github.com/elementary/gala/pull/760
* https://github.com/elementary/greeter/pull/423

worldofpeace committed:
pantheon.wingpanel: 2.3.0 -> 2020-04-04

support 3.36

worldofpeace committed:
pantheon.elementary-settings-daemon: 3.34.1 -> 3.36.0

worldofpeace committed:
Revert "pantheon.elementary-settings-daemon: 3.34.1 -> 3.36.0"

Mutter 3.36 in gala is almost ready.

This reverts commit 7160b7c31c8a44ffdd0cd0f375624c2c9c834917.

worldofpeace committed:
gnome3.mutter334: init

Used for Pantheon

worldofpeace committed:
pantheon.mutter: 328 -> 334

worldofpeace commented:
@GrahamcOfBorg test pantheon

jtojnar commented on pkgs/desktops/pantheon/desktop/gala/default.nix:
```diff
@@ -70,6 +70,7 @@ stdenv.mkDerivation rec {
 
   patches = [
     ./plugins-dir.patch
+    ./use-new-notifications-default.patch
```
Can't we use the override here as well?

worldofpeace replied:
It was weird. When gala would crash it would revert to not using the new notifications if I didn't do this.

jtojnar replied:
Yeah, it will need to be done here too since if gala somehow loses the environment, it will fall back to the one from wrappers. But I meant that we could try to re-use the overrides, since that is what they are for – affecting the default value in the compiled schemas.

worldofpeace replied:
Ahh, you mean put `use-new-notifications` in `20_elementary.pantheon.wm.gschema.override`.?

jtojnar replied:
Possibly. Though it probably does not matter.

jtojnar commented on pkgs/desktops/pantheon/default.nix:
```diff
@@ -110,7 +110,9 @@ lib.makeScope pkgs.newScope (self: with self; {
   # Take note of "I am holding off on "fixing" this bug for as long as possible."
-  elementary-settings-daemon = callPackage ./services/elementary-settings-daemon { };
+  elementary-settings-daemon = callPackage ./services/elementary-settings-daemon {
+    inherit (gnome3) gnome-desktop;
```
Ugh, we should move this to top-level as well.

jtojnar commented on pkgs/desktops/pantheon/services/elementary-notifications/default.nix:
```diff
@@ -0,0 +1,59 @@
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkgconfig
```
In new derivations we should use

```suggestion
    pkg-config
```

jtojnar commented on pkgs/desktops/pantheon/services/elementary-notifications/default.nix:
```diff
@@ -0,0 +1,59 @@
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Gtk notification server for Pantheon";
```
```suggestion
    description = "GTK notification server for Pantheon";
```

jtojnar commented on pkgs/desktops/pantheon/services/elementary-notifications/default.nix:
```diff
@@ -0,0 +1,59 @@
+    python3
+    vala
+    wrapGAppsHook
+  ];
```
```suggestion
    glib # for glib-compile-schemas
  ];
```

jtojnar commented on pkgs/desktops/pantheon/desktop/elementary-session-settings/default.nix:
```diff
@@ -80,9 +81,20 @@ stdenv.mkDerivation rec {
+  ];
+
   nativeBuildInputs = [
+    elementary-settings-daemon
```
??

worldofpeace replied:
my patch needs it https://github.com/elementary/session-settings/commit/39918f4ec64fa9ed5affa109d6a692b97ae4ff01.patch. version compare with the info from pkgconfig.
I consider that native

jtojnar replied:
We are asking what version will be installed so I would still consider it a run-time dependency – though in this case either will work.

worldofpeace committed:
pantheon.wingpanel-indicator-notifications: fix do not disturb

worldofpeace committed:
pantheon.switchboard-plug-notifications: fix do not disturb

worldofpeace commented:
Found some bugs with do not disturb not working with the new notification server:
https://github.com/elementary/switchboard-plug-notifications/pull/66
https://github.com/elementary/wingpanel-indicator-notifications/pull/110 (merged)

They were using `lookup` from `SettingsSchemaSource get_default`, but without `recursive` being true.
Reading the docs that is going to always be a problem on NixOS:
> The returned source may actually consist of multiple schema sources from different directories, depending on which directories were given in `XDG_DATA_DIRS` and `GSETTINGS_SCHEMA_DIR`. For this reason, all lookups performed against the default source should probably be done recursively.

worldofpeace commented:
I opened a bunch more:
https://github.com/elementary/switchboard-plug-keyboard/pull/278
https://github.com/elementary/greeter/pull/425
https://github.com/elementary/onboarding/pull/91
https://github.com/elementary/calendar/pull/502
https://github.com/elementary/switchboard-plug-locale/pull/119

worldofpeace commented:
Woah, thank you @tintou for merging all of those PRs so quickly :sparkles:

worldofpeace commented:
I'm running this just fine. I think I can merge this.
(I wish I could make an exception for 20.03, but I don't know how stable it will be just yet)
